internal/host/host.go

```diff
@@ -1412,3 +1417,49 @@ func (m *Manager) HandleReclaimBootArtifactDownload(ctx context.Context, h *mode
 func (m *Manager) HandleReclaimFailure(ctx context.Context, h *models.Host) error {
 	return m.sm.Run(TransitionTypeReclaimFailed, newStateHost(h), &TransitionArgsUnbindHost{ctx: ctx, db: m.db})
 }
+
+// autorenameHost renames a host if it is necessary and possible. For example, if the name is
+// localhost.localdomain and the MAC address of the main network interface is A8:CD:16:AE:79:01 it
+// will be renamed to a8-cd-16-ae-79-01. Returns a boolean indicating if the host was actually
+// renamed.
+func (m *Manager) autorenameHost(host *models.Host, inventory *models.Inventory) bool {
+	if host.RequestedHostname != "" {
+		return false
+	}
+	if inventory == nil || inventory.Hostname == "" {
+		return false
+	}
+	if !funk.Contains(hostutil.ForbiddenHostnames, inventory.Hostname) {
```

Comment:
Suggestion to use the `ValidateHostname` function here https://github.com/openshift/assisted-service/blob/3f535bd36578f5b1a5576610f19df4daee541138/internal/host/hostutil/host_utils.go#L80

and return false if err != nil 

Since the function has all the validations for a valid hostname.

Reply:
@CrystalChun the intent here is to automatically rename the host only if it is using a forbidden host name, not if it using an incorrect host name. For example, `my host` is an incorrect host name but we don't want to automatically replace it with the name calculated from the MAC address (or so I understood). If we use that `ValidateHostname` function here then we will be doing the rename for both forbidden and incorrect host names.

Reply:
Oh gotcha! How come we're only renaming forbidden and not incorrect host names?

Reply:
That is my interpretation of the ticket: https://issues.redhat.com/browse/MGMT-8460. I assume that we only want to do this automatic renaming when the user didn't explicitly set a name for the host. I may be wrong. Can you take a look at that ticket and share your thoughts?

Reply:
I see, I do agree with the assumption to set a name when the user didn't explicitly set a name. 

However, I'm wondering if it should automatically change the hostname when the inventory hostname is:
1. just forbidden (like in this check) 
2. forbidden AND invalid (e.g. `my host` or a hostname > 64 characters)

I'm not too sure which experience would be better for the user. 

E.g. If they have a host that doesn't have a forbidden name but the hostname is invalid (>64 characters) would it be better to warn the user, change it automatically, or something else?

Reply:
@CrystalChun I'd say that would want to automatically rename the host only when it is `localhost` (or any of its variants). My reasoning is that when the host name is `localhost` it most probably means that the user didn't make any effort to assign a name to the host. In cases like `my host` or longer than 64 characters it means that the user made the effort, but incorrectly, so we would want to tell her, and her to fix it, like we do today.

Reply:
That's a good point to only assign the host name when the user hasn't attempted to change it. I agree with you! Thanks for explaining!